yt_dlp/extractor/gedidigital.py

```diff
@@ -1,198 +1,104 @@
-import re
+import json
 
 from .common import InfoExtractor
-from ..utils import (
-    base_url,
-    determine_ext,
-    int_or_none,
-    url_basename,
-    urljoin,
-)
+from ..utils import int_or_none, js_to_json
 
 
 class GediDigitalIE(InfoExtractor):
-    _VALID_URL = r'''(?x:(?P<base_url>(?:https?:)//video\.
+    _VALID_URL = r'''(?x:(?:https?:)//(?:\w+\.)?
         (?:
-            (?:
-                (?:espresso\.)?repubblica
-                |lastampa
-                |ilsecoloxix
-                |huffingtonpost
-            )|
-            (?:
-                iltirreno
-                |messaggeroveneto
-                |ilpiccolo
-                |gazzettadimantova
-                |mattinopadova
-                |laprovinciapavese
-                |tribunatreviso
-                |nuovavenezia
-                |gazzettadimodena
-                |lanuovaferrara
-                |corrierealpi
-                |lasentinella
-            )\.gelocal
-        )\.it(?:/[^/]+){2,4}/(?P<id>\d+))(?:$|[?&].*))'''
-    _EMBED_REGEX = [rf'''(?x)
-            (?:
-                data-frame-src=|
-                <iframe[^\n]+src=
-            )
-            (["'])(?P<url>{_VALID_URL})\1''']
```

Comment:
Shouldn't we try to fix embed support instead of removing it altogether?

Reply:
iirc the new embed urls are compatible with the page regex, I'll add a test to see if there are any problems

Reply:
Ah, so does it mean that with this `_EMBED_REGEX` (that I deleted) it's possible to also download the video if embedded in other sites? If that's the case it can probably be restored like it was

Reply:
Yep, that's the idea. You'll probably also need to restore `_extract_embed_urls` and `_sanitize_url`, as well as tweaking the `_VALID_URL` since `_EMBED_REGEX` uses it as a fullmatch and we need to allow for query strings etc (see what the old pattern was doing)